hydromt_sfincs/sfincs.py

```diff
@@ -2600,6 +2604,208 @@ def setup_wind_forcing(self, timeseries=None, magnitude=None, direction=None):
+        * **wave_period** forcing: peak wave period time series [s]
+        * **wave_direction** forcing: mean wave direction time series [coming from degree wrt North]
+        * **wave_spreading** forcing: wave spreading time series [degree]            
+
+        Parameters
+        ----------
+        geodataset: str, Path, xr.Dataset, optional
+            Path, data source name, or xarray data object for geospatial point timeseries.
+        timeseries: str, Path, pd.DataFrame, optional
+            Path, data source name, or pandas data object for tabular timeseries.
+        locations: str, Path, gpd.GeoDataFrame, optional
+            Path, data source name, or geopandas object for bnd point locations.
+            It should contain a 'index' column matching the column names in `timeseries`.
+        buffer: float, optional
+            Buffer [m] around model wave boundary cells to select wave gauges,
+            by default 5 km.
+        merge : bool, optional
+            If True, merge with existing forcing data, by default True.
+
+        See Also
+        --------
+        set_forcing
+        """
+        gdf_locs, df_ts = None, None
+        
+        # If we wouldn't do any clipping, but just parse a geodataset directly:
+        # self.set_forcing(geodataset, name="snapwave", split_dataset=False)
+
+        # TODO: add this back in later
+        tstart, tstop = self.get_model_time()  # model time
+        # buffer around snapwave msk==2 values
+        
+        # if np.any(self.snapwave_msk == 2):
+        #     region = self.mask.where(self.snapwave_msk == 2, 0).raster.vectorize() 
+        #     #TODO: check if will be 'snapwave_msk' or 'wave_mask' or 'snapwave.msk' or ...
+        # else:
+        #     region = self.region
+
+        #read wave data from geodataset or geodataframe
+        region = self.region
+        
+        if geodataset is not None:
+            # read and clip data in time & space
+            da = self.data_catalog.get_geodataset(
+                geodataset,
+                geom=self.region,
+                buffer=buffer,
+                variables=["hs", "tp", "wd", "ds"], #TODO: Question - is this correct? will the data_catalog then load all 4 required vars?
+                time_tuple=(tstart, tstop),
+                 crs=self.crs,
+            )
+            # df_ts = da.transpose(..., da.vector.index_dim).to_dataframe()
+            gdf_locs = da.vector.to_gdf()
+            gdf_locs = gdf_locs.to_crs(self.crs) 
+            gdf_locs2 = gdf_locs.copy()
+            gdf_locs2 = gdf_locs2.to_crs(self.crs)
+ 
+            df_hs = da['hs'].transpose(..., da.vector.index_dim).to_pandas()  
+            df_tp = da['tp'].transpose(..., da.vector.index_dim).to_pandas()         
+            df_dir = da['wd'].transpose(..., da.vector.index_dim).to_pandas()         
+            df_ds = da['ds'].transpose(..., da.vector.index_dim).to_pandas()         
```

Comment:
Since the crs=self.crs does not seem to work, I think we can replace this for:

```suggestion
            ds = self.data_catalog.get_geodataset(
                geodataset,
                geom=self.region,
                buffer=buffer,
                variables=["hs", "tp", "wd", "ds"], 
                time_tuple=(tstart, tstop),
            )
     
            gdf = ds.vector.to_gdf()
            gdf = gdf.to_crs(self.crs)
            ds = GeoDataset.from_gdf(gdf, ds, index_dim=gdf.index.name)
```

Note, this does require the variables in the input dataset to be names hs, tp, wd and ds. The data_catalog could be used to rename those